docs/advanced-usage/compute-backends/kubernetes/index.md

```diff
@@ -108,3 +100,87 @@ You can set `kubernetes_memory_limit` in every `rule` under `resources`.
         "docker://python:2.7-slim"
     ...
 ```
+
+## Job timeout
+
+When the job exceeds the specified time limit, it will be terminated by Kubernetes and marked as failed.
+
+To set the job timeout, you can specify `kubernetes_job_timeout` in the specification of **each workflow step**.
+Time is measured in **seconds**.
+
+Read more about the expected values and timeout logic on [Kubernetes official documentation](https://kubernetes.io/docs/concepts/workloads/controllers/job/#job-termination-and-cleanup)
+
+You can configure the `steps` in the respective specifications for Serial, Yadage, CWL and Snakemake workflow engines.
+
+For **Serial**, you can set `kubernetes_job_timeout` in every `step` of workflow specification.
+
+```yaml hl_lines="6"
+   ...
+   steps:
+      - name: reana_demo_helloworld_job_timeout
+        environment: 'python:2.7-slim'
+        compute_backend: kubernetes
+        kubernetes_job_timeout: 60
+        commands:
+            - python helloworld.py
+```
+
+For **Yadage**, you can set `kubernetes_job_timeout` in every `step` under `environment.resources`.
+
+```yaml hl_lines="19"
+   ...
+   stages:
+   - name: reana_demo_helloworld_job_timeout
+      dependencies: [init]
+      scheduler:
+        scheduler_type: 'singlestep-stage'
+        parameters:
+          helloworld: {step: init, output: helloworld}
+        step:
+          process:
+            process_type: 'string-interpolated-cmd'
+            cmd: 'python "{helloworld}"'
+          environment:
+            environment_type: 'docker-encapsulated'
+            image: 'python'
+            imagetag: '2.7-slim'
+            resources:
+              - compute_backend: kubernetes
+              - kubernetes_job_timeout: 60
+```
+
+For **CWL**, you can set `kubernetes_job_timeout` in every `step` under `hints.reana`.
+
+```yaml hl_lines="7"
+   ...
+   steps:
+    first:
+      hints:
+        reana:
+          compute_backend: kubernetes
+          kubernetes_job_timeout: 60
+      run: helloworld_job_timeout.tool
+      in:
+        helloworld: helloworld_job_timeout
+      out: [result]
+```
+
+For **Snakemake**, you can set `kubernetes_job_timeout` in every `rule` under `resources`.
+
+```yaml hl_lines="12"
+  ...
+  rule helloworld:
+    input:
+        helloworld=config["helloworld"],
+        inputfile=config["inputfile"],
+    params:
+        sleeptime=config["sleeptime"]
+    output:
+        "results/greetings.txt"
+    resources:
+        compute_backend="kubernetes",
+        kubernetes_job_timeout=60
```

Comment:
Hmm, I've been quoting the Snakefile value in my local tests `kubernetes_job_timeout="60"` :wink: but the integer is better.